src/load-modules.ts

```diff
@@ -43,9 +43,9 @@ export type NameFormatter = (
 /**
  * Dependencies for `loadModules`
  */
-export interface LoadModulesDeps {
+export interface LoadModulesDeps<T extends object> {
```

Comment:
`loadModules` is dynamic already, I don't think we need to worry about it here.

Reply:
Don't need to worry in this file altogether? or just this interface and its uses?

Reply:
I think the entire file. Loading modules from the file system can never provide any static type analysis.

Reply:
Reverted this file